pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2971,6 +2998,11 @@ public void revertReplaceSegments(String tableNameWithType, String segmentLineag
           // routing table because it is possible that there has been no EV change but the routing result may be
           // different after updating the lineage entry.
           sendRoutingTableRebuildMessage(tableNameWithType);
+
+          // Invoke the proactive clean-up for segments that we no longer needs in case 'forceRevert' is enabled
+          if (forceRevert) {
+            deleteSegments(tableNameWithType, lineageEntry.getSegmentsTo());
```

Comment:
Currently we don't have any cleanup logic for `REVERTED` lineage entry. Also we are removing the segments asynchronously here, it's possible that some segments in `segmentsTo` are not deleted successfully. We can handle the cleanup for both in retention manager in following prs.

Reply:
We do have the clean-up logic for `REVERTED` entries in the retention manager. I added it as part of the revert API PR. https://github.com/apache/pinot/pull/7662/files#diff-fc6f837f3641465c40ccf44fdded9bd37a3c7c9fb966d4710b2b034da5844759R232

Reply:
Oh, I missed that, thanks!